gate-web/src/main/groovy/com/netflix/spinnaker/gate/services/SlackService.groovy

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright 2019 Netflix, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.spinnaker.gate.services
+
+import com.fasterxml.jackson.annotation.JsonIgnoreProperties
+import retrofit.http.GET
+import retrofit.http.Query
+
+interface SlackService {
+  @GET('/conversations.list?limit=1000&exclude_archived=true&pretty=1') // https://api.slack.com/methods/conversations.list
+  SlackChannelsResult getChannels(@Query("token") String token, @Query("cursor") String cursor)
```

Comment:
The linked slack docs say: 
```
Pagination
This method uses cursor-based pagination to make it easier to incrementally collect information. To begin pagination, specify a limit value under 1000. We recommend no more than 200 results at a time.
```
Should we heed that advice? (I know nothing about the slack api)

Reply:
I was worried about slack's rate limiting here, but I will see how much I can reduce the limit to.